### ISSUE TITLE
Please add info on Accepting-Suggestions

### ISSUE THREAD

**Issue by Jay-o-Way** [check-spelling/check-spelling#26]
I'm reading https://github.com/check-spelling/check-spelling/wiki/Accepting-Suggestions#windows and I still don't know what to do. How exactly do you "Run" a file inside of a repository? And how do I know if it worked?

**Comment by jsoref** [comment]
Thanks for the feedback. If the perl command worked,  `git status` should show changes to the check spelling metadata files (which files that is will depend on the configuration). It doesn't create the commit, so you'll have to create a commit as normal.

Run a file means that your command prompt/terminal's current directory should be the top level directory of the git repository and you'd create a file with the perl command and then use `perl script.pl`   where `script.pl` is a file you've created with the contents of the quoted text from the comment. I understand that's pretty complicated.

As an alternative to that, I've extended the text with information on how to use Git Bash – I intend to add a picture to make it easier for people to recognize/understand what Git Bash is, but basically when you install Git for Windows, it adds a start menu item called "Git Bash".

Please let me know if this helps. The wiki is designed to be editable and I'm trying to evolve it.

**Comment by Jay-o-Way** [comment]
### A bit of background
I have been helping with PowerToys for quite a time now, but I had to learn everything "on the go":

- Github workflow (like issues, pr's, file structure; cloning, forking & branching...)
  - I still hardly know what "checking in/out" means
- All the different languages and terms that are used (XAML, Islands, WinUI[3], WPF, C#, C++ and so much more 🤯 )
- Visual Studio (good lord, that thing is powerful but complicated and - for my pc - a heavy program!)
  - and combining this with the above...
- GitHub Desktop (nice to manage pr's and stuff)

Right now I haven't even installed Visual Studio yet because of the weight of it. Recently using Visual Code instead, which is obviously less powerful/integrated too. And now I see I might need another one: Git.

### Having that said / Moving back to the issue
I think it would be nice to look at the document with "fresh eyes". For example: there are many ways to approach even the "run a file" thing: CMD, PowerShell or one of the different editor's built-in Terminal? I have tried several ways, and also tried several versions of the copied text: full text or without some of the quote characters. Didn't see any actual result, not even a changed `.github/actions/spell-check/expect.txt` file.

When we get a message inside a pr that there is a spelling issue ~(the amount of them can get annoying, but that's a different point)~ why isn't there simply a **button** to run these things automatically, server-side? 🙁

**Comment by jsoref** [comment]
I've updated the text after spending a bit of time w/ Windows 7 to try to make things work.

Let me know if it's better/works well enough to use.

It's definitely a work in progress, and it was good for me to try it now, as I could not get a feature I recently added to work w/ Windows (`rsync`).

**Comment by Jay-o-Way** [comment]
Thanks. I see alot of information and most of it is still gibberish to me. Posix shells? Git bash? JQ? And using this party software?
Message like "If the command includes `rsync`, it probably won't work on Windows..." also make me very sad 😐

P.S. Windows 7?

**Comment by jsoref** [comment]
Windows 7 is available from Microsoft via https://modern.ie/ -- https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/ -- it's handy for those of us who don't natively run Windows. It's also small enough to download w/o crying.

I don't think people will run into the `rsync` case, it was used for an experimental feature in a limited set of alpha releases for 0.0.20. I could hide the item (and probably will once I release 0.0.20).

---

Third party software ... Well... The easiest thing to do is to use the shinier model in your personal fork.

Here's a sample of how that would work:
https://github.com/check-spelling/PowerToys/pull/7#issuecomment-1113772200

Note that before commenting, I ran:
https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key#script
... which added the secret to make it work smoothly. It's _suppposed_ to work without that, but in my testing earlier this afternoon it didn't. I'll try to look into it shortly.

Basically, then have an updated branch https://github.com/check-spelling/PowerToys/commits/update-demo from which you could make a PR or pull to locally use...

**Comment by Jay-o-Way** [comment]
Straight up question: Can't this (spell check update) be done server-side with the push of a button?

**Comment by Jay-o-Way** [comment]
General support for Win.7 ended in jan 2015 and extended support ended in jan 2020. I don't think anyone (especially developers) should use that as a model any more. But hey, that's my opinion...

**Comment by jsoref** [comment]
> Straight up question: Can't this (spell check update) be done server-side with the push of a button?

TL;DR there are lots of ways it _could_ be done, and each way is its own trail of tears. I've tried a bunch of them and the bot has support for a bunch of them to the best of its abilities.

There are a bunch of variables...
* Some users don't like having their branches changed -- this means that it needs some level of opt-in
   * for this today, I'm using a comment handler as you can see in https://github.com/check-spelling/PowerToys/pull/7#issuecomment-1113772200
   * In theory, GitHub has a feature of check suites that allows one to off an action, but afaict that isn't available to the github-actions -- afaict, using this feature would require me to run a web server that's ever on which is something I'm trying to avoid.
* Neither the github token nor a deploy key have the ability to push to forks -- see https://github.com/github/docs/issues/15760
* I've asked about an alternative approach for updating across forks: https://github.com/github/feedback/discussions/14216
* If not pushing across forks...
  * The default bot token won't trigger new workflows which means that if it does push, it won't have a passing check unless someone commits again. (And the last time I tried this, it didn't work at all, I can't tell if GitHub broke something else.)
  * There are provisions for a fancier token, and they work (see https://github.com/check-spelling/PowerToys/pull/7/commits/5d45e9dede6a7aabb41b24857d8644f9cfd4c1d5), but they do require repository owners to set up a [deploy key](https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key)

**Comment by jsoref** [comment]
There's actually a [button coming!](https://github.com/check-spelling/check-spelling/wiki/Feature:-Update-with-deploy-key)

Collapse comments is [also coming!](https://github.com/check-spelling/check-spelling/wiki/Feature:-Collapse-older-check-comments)

I've been trying to ship a release for probably 9 months, but I keep tinkering with things. I have one thing I noticed reecntly that I need to fix. (Possibly fixed -- this comment got stuck.)

You can see both of those features at play in this sample PR:
https://github.com/jsoref/examples-testing/pull/11

---

I know there are a whole bunch of ways to run things on Windows. It's partially why I hadn't tried to write text for it for so long. I've been mostly hoping that I'd get the server side solution ready to ship and that I could convince people to use it.

I _think_ what I'll probably do is ask the microsoft repositories to configure the workflow w/ the feature, but not include a secret, that would let people use the feature on forks while giving the microsoft folks time to look it over before deploying a key for the microsoft repositories. To do that, I'll probably have to add a bit more code to make the feature a tri-state (off, on-if-secret-available, on) instead of a boolean (off, on w/ less friendly experience if secret isn't available).

It's mostly ok if someone doesn't remove previously acknowledged words as long as someone does eventually before someone unintentionally reintroduces something that should no longer be used.

* Imagine there's 1 instance of someone's name misspelled (because it wasn't caught in review).
* Someone creates a branch "cleanup" that fixes the misspelling.
* They create a PR for "cleanup"
* Someone creates a branch "more" (before "cleanup" merges)  and copies the misspelling elsewhere.
* If the "cleanup" PR is merged and doesn't remove the misspelling, then the word will be tolerated.
* Someone creates a PR for "more". Because it's tolerated, people don't realize it's a typo and shouldn't be added.

This is just a race condition (I managed to file a bug about a race to MS last night, everyone has them), and there are other flavors relating to this project, it's ok. The hope is eventually the cleanup will stick.

**Comment by jsoref** [comment]
v0.0.20 was released. With a workflow refresh (I'll try to send PRs shortly), collapse comments will be available.

**Comment by jsoref** [comment]
As of [0.0.21](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.21), the command is now just `curl`+`perl` and should work well on Windows. (Technically it needs `gh` and `git` as well, but it should walk you through those.)

I believe the PowerToys repository should be using it. Please give it a try and let me know if you have problems.
